packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Primitives/CodeWriterDeclaration.cs

```diff
@@ -19,24 +36,35 @@ public CodeWriterDeclaration(string name)
 
         public string RequestedName { get; }
 
-        public string ActualName => _actualName ?? _debuggerName ?? throw new InvalidOperationException($"Declaration {RequestedName} is not initialized");
+        internal string GetActualName(CodeWriter.CodeScope scope)
+        {
+            if (!_actualNames.TryGetValue(scope, out var actualName))
+                return _debuggerName ?? throw new InvalidOperationException($"Declaration {RequestedName} is not initialized");
+
+            return actualName;
+        }
 
-        internal void SetActualName(string? actualName)
+        internal void SetActualName(string? actualName, CodeWriter.CodeScope scope)
         {
-            if (_actualName != null && actualName != null)
+            var isScopeDeclared = _actualNames.ContainsKey(scope);
+            if (isScopeDeclared && actualName != null)
             {
-                throw new InvalidOperationException($"Declaration {_actualName} already initialized, can't initialize it with {actualName} name.");
+                throw new InvalidOperationException($"Declaration {_actualNames[scope]} already initialized, can't initialize it with {actualName} name.");
             }
 
-            _actualName = actualName;
+            if (actualName is not null)
+            {
+                _actualNames[scope] = actualName;
+            }
+            else if (isScopeDeclared)
+            {
+                _actualNames.Remove(scope);
+            }
         }
 
         internal void SetDebuggerName(string? debuggerName)
         {
-            if (_actualName == null)
-            {
-                _debuggerName = debuggerName;
-            }
+            _debuggerName = debuggerName;
```

Comment:
kind of curious, where this is used?
This class does not have a debugger display.
Is this used in like expressions?

Reply:
I actually don't know, let me see if I can remove it